react-validated-input-component/my-app/src/ValidatedInput.js

```diff
@@ -0,0 +1,61 @@
+import { useState } from 'react';
+
+export default function ValidatedInput() {
+  const [pw, setPw] = useState('');
+  const [isValid, setIsValid] = useState(false);
+  const [attempted, setAttempted] = useState(false);
```

Comment:
I'm not convinced this is necessary. `err` will always be empty until `validate` is called, and it won't get called until the user starts typing. So instead of checking `attempted` you could just check if `err` is empty.